app/mailers/custom_devise_mailer.rb

```diff
@@ -7,6 +7,8 @@ def subject_for(key)
     # Replace the invitation instruction subject for partner users
     # that were invited by other partner users.
 
+    resource.reload
+
     if resource.is_a?(PartnerUser) && resource.id == resource.partner.primary_user&.id
```

Comment:
The problem is partner primary user is nil, when test in this condition.

Before to check this condition is necessary update resource data.